salt/_modules/metalk8s_volumes.py

```diff
@@ -180,7 +182,8 @@ def format(self, force=False):
         """
         # Check that the backing device is not already formatted.
         # Bail out if it is: we don't want data loss because of a typo…
-        if _get_from_blkid(self.path, 'TYPE'):
+        device_info = _get_from_blkid(self.path)
+        if device_info.fstype or device_info.has_partition:
             raise Exception('backing device `{}` already formatted'.format(
```

Comment:
Exception no longer matches the causes for errors. Split them up in 2 cases with an appropriate error message.

Reply:
Having a FS or a partition table are both a proof that the device is already formatted, no?
Do we really want to make a distinction here?

Reply:
Yes.

Either it's a device that's already formatted but not 'by us', so there's a 'conflict'.

Or, the device path we got (e.g. in case of a `rawBlockDevice`) is a block device that contains partitions. As such this is more likely a configuration error (someone provided `/dev/sda` whilst it should've been `/dev/sda1`, for example).

Granted, in both cases there's nothing *we* can do except for raising an error. However, the root cause why such exception is raised is quite different, and pointing the user in the right direction ('Already formatted' vs. 'Contains partitions, not something we can format') is useful.

Reply:
Indeed, will do.